metricbeat/module/kubernetes/container/container.go

```diff
@@ -83,5 +87,13 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 		return nil, err
 	}
 
+	m.enricher.Enrich(events)
+
 	return events, nil
 }
+
+// Close stops this metricset
+func (m *MetricSet) Close() error {
```

Comment:
Did you test if Close is called?

Reply:
Yes, surprisingly it's called :smiley_cat: here: https://github.com/elastic/beats/blob/master/metricbeat/mb/module/wrapper.go#L132